### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by FuriousGopher**

### CODE REVIEW

File: src/send-games-data/utils/send-game-data.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+export const sendGameData = async () => {
+  try {
+    const data = await fsPromises.readFile(
+      'src/send-games-data/games-info/game-data.json',
+    );
+    return JSON.parse(data.toString());
+  } catch (err) {
+    throw err;
+  }
+};
```
Comment (DavidSurkov) [inline src/send-games-data/utils/send-game-data.ts]: `export const sendGameData = async (searchQuery?: string) => {
  try {
    const data = await fsPromises.readFile(
      'src/send-games-data/games-info/game-data.json',
    );
const parsedGamesArray = JSON.parse(data.toString());
if (searchQuery) {
return parsedGamesArray.filter((game) =>
        game.title.toLowerCase().includes(searchQuery.toLowerCase()),
      )
}
    return JSON.parse(data.toString());
  } catch (err) {
    throw err;
  }
};`

File: src/send-games-data/send-games-info.controller.ts [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
-  async getGameData(@Res() res: Response) {
-    return fs.readFile(
-      'src/send-games-data/games-info/game-data.json',
-      (err, data) => {
-        if (err) throw err;
-        res.send(JSON.parse(data.toString()));
-      },
-    );
+  async getGameData(@Res() res: Response, @Query('title') title?: string) {
+    const gameData = await sendGameData();
```
Comment (DavidSurkov) [inline src/send-games-data/send-games-info.controller.ts]: fetchGameData

**Review by DavidSurkov — CHANGES_REQUESTED** [review]

File: src/send-games-data/send-games-info.controller.ts [hunk 3]
```diff
@@ -6,16 +6,8 @@ import { sendGameData } from './utils/send-game-data';
 export class SendGamesInfoController {
   @Get()
   async getGameData(@Res() res: Response, @Query('title') title?: string) {
-    const gameData = await sendGameData();
+    const gameData = await sendGameData(title);
```
Comment (DavidSurkov) [inline src/send-games-data/send-games-info.controller.ts]: THIS function is not SENDING DATA it's fetching it from file. WHY THE NAME DOES NOT REPRESENT ITS FUNCTION??

**Review by DavidSurkov — CHANGES_REQUESTED** [review]
FIX

File: src/send-games-data/send-games-info.controller.ts (hunk 3 above)
Comment (FuriousGopher) [reply src/send-games-data/send-games-info.controller.ts]: i go to HR

File: src/send-games-data/send-games-info.controller.ts [hunk 4]
```diff
@@ -6,8 +6,8 @@ import { sendGameData } from './utils/send-game-data';
 export class SendGamesInfoController {
   @Get()
   async getGameData(@Res() res: Response, @Query('title') title?: string) {
-    const gameData = await sendGameData(title);
+    const fetchGameData = await sendGameData(title);
```
Comment (DavidSurkov) [inline src/send-games-data/send-games-info.controller.ts]: ebobo?

**Review by DavidSurkov — CHANGES_REQUESTED** [review]

**Review by DavidSurkov — APPROVED** [review]

**Pull request merged**